packages/eas-cli/src/commands/workflow/status.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * EAS Workflow Status Command
+ *
+ * This command shows the status of an existing workflow run.
+ *
+ * If no run ID is provided, you will be prompted to select from recent workflow runs for the current project.
+ *
+ * If the selected run is still in progress, the command will show the progress of the run, with an option
+ * to show periodic status updates while waiting for completion (similar to `eas workflow:run --wait`).
+ *
+ */
+
+import { Flags } from '@oclif/core';
+import { boolish } from 'getenv';
+
+import { getWorkflowRunUrl } from '../../build/utils/url';
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASNonInteractiveFlag, EasJsonOnlyFlag } from '../../commandUtils/flags';
+import {
+  choiceFromWorkflowRun,
+  processWorkflowRuns,
+  showWorkflowStatusAsync,
+  workflowRunExitCodes,
+} from '../../commandUtils/workflow/utils';
+import { WorkflowRunStatus } from '../../graphql/generated';
+import { AppQuery } from '../../graphql/queries/AppQuery';
+import { WorkflowRunQuery } from '../../graphql/queries/WorkflowRunQuery';
+import Log, { link } from '../../log';
+import { getOwnerAccountForProjectIdAsync } from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export default class WorkflowStatus extends EasCommand {
+  static override description =
+    'show the status of an existing workflow run. If no run ID is provided, you will be prompted to select from recent workflow runs for the current project.';
+
+  static override args = [
+    {
+      name: 'workflowRunId',
```

Comment:
```suggestion
      name: 'WORKFLOW_RUN_ID',
```
? that `WORKLFOWRUNID` from PR description is kinda weird. Inspired by https://github.com/expo/eas-cli/blob/b8a056b43e242c87e2cbc19b9d94e6c85e9517dc/packages/eas-cli/src/commands/deploy/delete.ts#L39